# Write for humans 

Write variables with more than 2 characters. Remember your code is read by a human not a machine. (write-for-humans)

## Rule Details

This rule aims to assist you in writing code that can be read and understood better by anyone who tries to maintain it (even for the author). Variable names that are more than 2 characters long is the absolute minimum.

Examples of **incorrect** code for this rule:

```js
var s = "Nope";
var ar = [];
```

Examples of **correct** code for this rule:

```js
var pageTitle = "Product page";
var products = [];
```

## When Not To Use It

When you are gonna be writing Mathematics.

## Further Reading

- Every goddamned article about writing good software.
